### PR TITLE
kernel: add driver support for aws variants in hybrid environments

### PR DESCRIPTION
**Pull request by markusboehme**
**Issue number:** n/a

**Description of changes:** Add driver support for using aws variants in hybrid environments. In particular, add support for booting from IDE/ATA devices and 5th generation Mellanox NICs.

**Testing done:** I collected kernel config diffs using `tools/diff-kernel-config`, build-testing relevant configurations. This is the overview:

```
config-aarch64-5.10-aws-k8s-1.23-diff:    0 removed,   0 added,   1 changed
config-aarch64-5.15-aws-dev-diff:         0 removed,  22 added,   9 changed
config-aarch64-5.15-metal-dev-diff:       0 removed,   0 added,   0 changed
config-x86_64-5.10-aws-k8s-1.23-diff:     0 removed,  16 added,   8 changed
config-x86_64-5.10-metal-k8s-1.23-diff:   0 removed,   0 added,   0 changed
config-x86_64-5.15-aws-dev-diff:          0 removed,  20 added,   9 changed
config-x86_64-5.15-metal-dev-diff:        0 removed,   0 added,   0 changed
```

Note that as expected only aws variants have their config changed. You can find a full config diff [in this gist](https://gist.github.com/markusboehme/5c713898631eb6d312ccf9c17a1ab429). Interesting to see that the 5.10 kernel for aarch64 almost had the desired configuration already. Feature selection for the mlx5 driver mirrors the existing configuration for metal variants.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by bcressey — APPROVED** [review]

**Review by zmrow — APPROVED** [review]
:night_with_stars:

File: packages/kernel-5.10/config-bottlerocket-aws [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Support boot from IDE disks
+CONFIG_ATA=y
+CONFIG_ATA_GENERIC=y
```
Comment (foersleo) [inline packages/kernel-5.10/config-bottlerocket-aws]: Do we want the generic driver? We do not ship it with the metal variants.
If we actually want it, we probably also want it on metal.

Comment (markusboehme) [reply packages/kernel-5.10/config-bottlerocket-aws]: On a second look, indeed, this won't be needed. I'll drop it in the next revision.

**Comment by markusboehme** [comment]
Dropped `CONFIG_ATA_GENERIC` for 2nd revision.

Refreshed summary of kernel config changes:

```
config-aarch64-5.10-aws-k8s-1.23-diff:    0 removed,   0 added,   0 changed
config-aarch64-5.15-aws-dev-diff:         0 removed,  22 added,   8 changed
config-aarch64-5.15-metal-dev-diff:       0 removed,   0 added,   0 changed
config-x86_64-5.10-aws-k8s-1.23-diff:     0 removed,  16 added,   7 changed
config-x86_64-5.10-metal-k8s-1.23-diff:   0 removed,   0 added,   0 changed
config-x86_64-5.15-aws-dev-diff:          0 removed,  20 added,   8 changed
config-x86_64-5.15-metal-dev-diff:        0 removed,   0 added,   0 changed
```

The full diff of changes can be seen in [this gist](https://gist.github.com/markusboehme/0133426ad6eca41f9012901b2839c965).

**Review by jpculp — APPROVED** [review]

**Review by foersleo — APPROVED** [review]

**Pull request merged**